Exception occurred after connecting to shared database suggesting a duplicate entry

### JabRef version

5.10 (latest release)

### Operating system

Windows

### Details on version and operating system

Windows 11 pro 64bit Version 22H2 Built 22621.2134

### Checked with the latest development build (copy version output from About dialog)

- [X] I made a backup of my libraries before testing the latest development version.
- [X] I have tested the latest development version and the problem persists

### Steps to reproduce the behaviour

1. Open JabRef
2. Connect to shared database
3. Fill required information
4. Click on Connect
5. An exception is thrown

![Capture d'écran 2023-09-06 140708](https://github.com/JabRef/jabref/assets/87969978/2ce6c937-eb94-4d4c-9eba-27ba936f0aae)


### Appendix

Following the error message, we checked the tables JABREF_ENTRY and ENTRY on DBeaver but didn't find a duplicate entry.
<details>


<summary>Log File</summary>

```
2023-09-06 14:02:43 [JavaFX Application Thread] org.jabref.gui.StateManager.setActiveDatabase()
INFO: No open database detected
2023-09-06 14:02:43 [JavaFX Application Thread] org.jabref.gui.theme.ThemeManager.updateThemeSettings()
INFO: Not updating theme because it hasn't changed
2023-09-06 14:02:43 [JavaFX Application Thread] org.jabref.gui.theme.ThemeManager.updateThemeSettings()
INFO: Theme set to Theme{type=DEFAULT, name=''} with base css StyleSheet{jrt:/org.jabref/org/jabref/gui/Base.css}
2023-09-06 14:04:12 [JavaFX Application Thread] org.jabref.logic.shared.DBMSSynchronizer.initializeDatabases()
INFO: Integrity check failed. Fixing...
2023-09-06 14:04:12 [JavaFX Application Thread] org.jabref.logic.shared.MySQLProcessor.setUp()
WARN: [VERSION_DB_STRUCT_DEFAULT] not Integer!
2023-09-06 14:04:12 [JavaFX Application Thread] org.jabref.logic.shared.MySQLProcessor.setUp()
INFO: Migrating from VersionDBStructure == 0
2023-09-06 14:04:12 [JavaFX Application Thread] org.jabref.logic.shared.DBMSSynchronizer.initializeDatabases()
ERROR: Could not check intergrity: java.sql.SQLIntegrityConstraintViolationException: (conn=4421) Duplicate entry '2' for key 'PRIMARY'
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.MariaDbStatement.executeInternal(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.MariaDbStatement.executeUpdate(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.logic.shared.MySQLProcessor.setUp(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.logic.shared.DBMSProcessor.setupSharedDatabase(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.logic.shared.DBMSSynchronizer.initializeDatabases(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.logic.shared.DBMSSynchronizer.openSharedDatabase(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.shared.SharedDatabaseUIManager.openNewSharedDatabaseTab(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.shared.SharedDatabaseLoginDialogViewModel.openSharedDatabase(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.shared.SharedDatabaseLoginDialogViewModel.openDatabase(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.shared.SharedDatabaseLoginDialogView.openDatabase(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.shared.SharedDatabaseLoginDialogView.lambda$new$0(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.util.ControlHelper.lambda$setAction$0(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventHandler$NormalEventFilterRecord.handleCapturingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventHandler.dispatchCapturingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventHandlerManager.dispatchCapturingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventHandlerManager.dispatchCapturingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventDispatcher.dispatchCapturingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.Node.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.Button.fire(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.behavior.ButtonBehavior.mouseReleased(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.inputmap.InputMap.handle(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.Scene$MouseHandler.process(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.Scene.processMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.Scene$ScenePeerListener.mouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.View.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.View.notifyMouse(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.win.WinApplication._enterNestedEventLoopImpl(Native Method)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.win.WinApplication._enterNestedEventLoop(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.Application.enterNestedEventLoop(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.EventLoop.enter(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.QuantumToolkit.enterNestedEventLoop(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.stage.Stage.showAndWait(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.HeavyweightDialog.showAndWait(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.Dialog.showAndWait(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.JabRefDialogService.showCustomDialogAndWait(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.shared.ConnectToSharedDatabaseCommand.execute(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.actions.JabRefAction.lambda$new$3(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.controlsfx.control.action.Action.handle(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.controlsfx.control.action.Action.handle(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.MenuItem.fire(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.ContextMenuContent$MenuItemContainer.doSelect(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.ContextMenuContent$MenuItemContainer.lambda$createChildren$12(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.Scene$MouseHandler.process(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.Scene.processMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.Scene$ScenePeerListener.mouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.View.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.View.notifyMouse(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException: Duplicate entry '2' for key 'PRIMARY'
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException.of(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(Unknown Source)
	... 113 more
Caused by: java.sql.SQLException: Duplicate entry '2' for key 'PRIMARY'
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(Unknown Source)
	... 114 more
2023-09-06 14:04:12 [JavaFX Application Thread] org.jabref.gui.FallbackExceptionHandler.uncaughtException()
ERROR: Uncaught exception occurred in Thread[#35,JavaFX Application Thread,5,main]: java.lang.IllegalStateException: java.sql.SQLIntegrityConstraintViolationException: (conn=4421) Duplicate entry '2' for key 'PRIMARY'
	at org.jabref@5.10.60000/org.jabref.logic.shared.DBMSSynchronizer.initializeDatabases(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.logic.shared.DBMSSynchronizer.openSharedDatabase(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.shared.SharedDatabaseUIManager.openNewSharedDatabaseTab(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.shared.SharedDatabaseLoginDialogViewModel.openSharedDatabase(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.shared.SharedDatabaseLoginDialogViewModel.openDatabase(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.shared.SharedDatabaseLoginDialogView.openDatabase(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.shared.SharedDatabaseLoginDialogView.lambda$new$0(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.util.ControlHelper.lambda$setAction$0(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventHandler$NormalEventFilterRecord.handleCapturingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventHandler.dispatchCapturingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventHandlerManager.dispatchCapturingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventHandlerManager.dispatchCapturingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventDispatcher.dispatchCapturingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.Node.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.Button.fire(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.behavior.ButtonBehavior.mouseReleased(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.inputmap.InputMap.handle(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.Scene$MouseHandler.process(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.Scene.processMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.Scene$ScenePeerListener.mouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.View.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.View.notifyMouse(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.win.WinApplication._enterNestedEventLoopImpl(Native Method)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.win.WinApplication._enterNestedEventLoop(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.Application.enterNestedEventLoop(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.EventLoop.enter(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.QuantumToolkit.enterNestedEventLoop(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.stage.Stage.showAndWait(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.HeavyweightDialog.showAndWait(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.Dialog.showAndWait(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.JabRefDialogService.showCustomDialogAndWait(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.shared.ConnectToSharedDatabaseCommand.execute(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.gui.actions.JabRefAction.lambda$new$3(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.controlsfx.control.action.Action.handle(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.controlsfx.control.action.Action.handle(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.control.MenuItem.fire(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.ContextMenuContent$MenuItemContainer.doSelect(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.scene.control.ContextMenuContent$MenuItemContainer.lambda$createChildren$12(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventHandler$NormalEventHandlerRecord.handleBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventHandler.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventHandlerManager.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.CompositeEventDispatcher.dispatchBubblingEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.BasicEventDispatcher.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventDispatchChainImpl.dispatchEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventUtil.fireEventImpl(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.event.EventUtil.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.event.Event.fireEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.Scene$MouseHandler.process(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.Scene.processMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/javafx.scene.Scene$ScenePeerListener.mouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler$MouseEventNotification.run(Unknown Source)
	at java.base/java.security.AccessController.doPrivileged(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler.lambda$handleMouseEvent$2(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.QuantumToolkit.runWithoutRenderLock(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.javafx.tk.quantum.GlassViewEventHandler.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.View.handleMouseEvent(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.View.notifyMouse(Unknown Source)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.win.WinApplication._runLoop(Native Method)
	at org.jabref.merged.module@5.10.60000/com.sun.glass.ui.win.WinApplication.lambda$runLoop$3(Unknown Source)
	at java.base/java.lang.Thread.run(Unknown Source)
Caused by: java.sql.SQLIntegrityConstraintViolationException: (conn=4421) Duplicate entry '2' for key 'PRIMARY'
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.createException(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.internal.util.exceptions.ExceptionFactory.create(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.MariaDbStatement.executeExceptionEpilogue(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.MariaDbStatement.executeInternal(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.MariaDbStatement.executeUpdate(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.logic.shared.MySQLProcessor.setUp(Unknown Source)
	at org.jabref@5.10.60000/org.jabref.logic.shared.DBMSProcessor.setupSharedDatabase(Unknown Source)
	... 109 more
Caused by: org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException: Duplicate entry '2' for key 'PRIMARY'
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.internal.util.exceptions.MariaDbSqlException.of(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.exceptionWithQuery(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.executeQuery(Unknown Source)
	... 113 more
Caused by: java.sql.SQLException: Duplicate entry '2' for key 'PRIMARY'
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readErrorPacket(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.readPacket(Unknown Source)
	at org.jabref.merged.module@5.10.60000/org.mariadb.jdbc.internal.protocol.AbstractQueryProtocol.getResult(Unknown Source)
	... 114 more
```

</details>

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -49,6 +49,7 @@ Note that this project **does not** adhere to [Semantic Versioning](https://semv
 - We fixed an issue where after creation of a new library, the new library was not focused. [koppor#592](https://github.com/koppor/jabref/issues/592)
 - We fixed an issue where double clicking on an url in the file field would trigger an exception instead of opening the browser [#10480](https://github.com/JabRef/jabref/pull/10480)
 - We fixed an issue where exporting "Embedded BibTeX pdf" without selecting an existing document would produce an exception. [#10101](https://github.com/JabRef/jabref/issues/10101)
+- We fixed an issue where it was not possible to connect to a shared database once a group with entries was added or other metadata modified [#10336](https://github.com/JabRef/jabref/issues/10336)
 
 ### Removed
 

--- a/src/main/java/org/jabref/logic/citationstyle/CitationStyle.java
+++ b/src/main/java/org/jabref/logic/citationstyle/CitationStyle.java
@@ -111,7 +111,7 @@ public class CitationStyle {
                 text = CSLUtils.readURLToString(url, StandardCharsets.UTF_8.toString());
             } else {
                 // if the url is null then the style is located outside the classpath
-                text = new String(Files.readAllBytes(Path.of(styleFile)), StandardCharsets.UTF_8);
+                text = Files.readString(Path.of(styleFile));
             }
             return createCitationStyleFromSource(text, styleFile);
         } catch (NoSuchFileException e) {

--- a/src/main/java/org/jabref/logic/shared/DBMSProcessor.java
+++ b/src/main/java/org/jabref/logic/shared/DBMSProcessor.java
@@ -53,20 +53,16 @@ public abstract class DBMSProcessor {
      * Scans the database for required tables.
      *
      * @return <code>true</code> if the structure matches the requirements, <code>false</code> if not.
-     * @throws SQLException
+     * @throws SQLException in case of error
      */
     public boolean checkBaseIntegrity() throws SQLException {
         boolean databasePassesIntegrityCheck = false;
         DBMSType type = this.connectionProperties.getType();
         Map<String, String> metadata = getSharedMetaData();
         if (type == DBMSType.POSTGRESQL || type == DBMSType.MYSQL) {
-            try {
-                Integer VERSION_DB_STRUCT = Integer.valueOf(metadata.get(MetaData.VERSION_DB_STRUCT));
-                if (VERSION_DB_STRUCT == getCURRENT_VERSION_DB_STRUCT()) {
-                    databasePassesIntegrityCheck = true;
-                }
-            } catch (Exception e) {
-                databasePassesIntegrityCheck = false;
+            int VERSION_DB_STRUCT = Integer.parseInt(metadata.get(MetaData.VERSION_DB_STRUCT));
+            if (VERSION_DB_STRUCT == getCURRENT_VERSION_DB_STRUCT()) {
+                databasePassesIntegrityCheck = true;
             }
         } else {
             databasePassesIntegrityCheck = checkTableAvailability("ENTRY", "FIELD", "METADATA");
@@ -117,7 +113,7 @@ public abstract class DBMSProcessor {
      * Creates and sets up the needed tables and columns according to the database type and performs a check whether the
      * needed tables are present.
      *
-     * @throws SQLException
+     * @throws SQLException in case of error
      */
     public void setupSharedDatabase() throws SQLException {
         setUp();
@@ -131,7 +127,7 @@ public abstract class DBMSProcessor {
     /**
      * Creates and sets up the needed tables and columns according to the database type.
      *
-     * @throws SQLException
+     * @throws SQLException in case of error
      */
     protected abstract void setUp() throws SQLException;
 
@@ -225,16 +221,15 @@ public abstract class DBMSProcessor {
                                            .map(BibEntry::getSharedBibEntryData)
                                            .map(SharedBibEntryData::getSharedID)
                                            .filter(id -> id != -1)
-                                           .collect(Collectors.toList());
+                                           .toList();
         if (localIds.isEmpty()) {
             return bibEntries;
         }
         try {
-            StringBuilder selectQuery = new StringBuilder()
-                    .append("SELECT * FROM ")
-                    .append(escape_Table("ENTRY"));
+            String selectQuery = "SELECT * FROM " +
+                    escape_Table("ENTRY");
 
-            try (ResultSet resultSet = connection.createStatement().executeQuery(selectQuery.toString())) {
+            try (ResultSet resultSet = connection.createStatement().executeQuery(selectQuery)) {
                 while (resultSet.next()) {
                     int id = resultSet.getInt("SHARED_ID");
                     remoteIds.add(id);
@@ -305,7 +300,7 @@ public abstract class DBMSProcessor {
      * Updates the whole {@link BibEntry} on shared database.
      *
      * @param localBibEntry {@link BibEntry} affected by changes
-     * @throws SQLException
+     * @throws SQLException in case of error
      */
     public void updateEntry(BibEntry localBibEntry) throws OfflineLockException, SQLException {
         connection.setAutoCommit(false); // disable auto commit due to transaction
@@ -313,7 +308,7 @@ public abstract class DBMSProcessor {
         try {
             Optional<BibEntry> sharedEntryOptional = getSharedEntry(localBibEntry.getSharedBibEntryData().getSharedID());
 
-            if (!sharedEntryOptional.isPresent()) {
+            if (sharedEntryOptional.isEmpty()) {
                 return;
             }
 
@@ -328,20 +323,19 @@ public abstract class DBMSProcessor {
                 insertOrUpdateFields(localBibEntry);
 
                 // updating entry type
-                StringBuilder updateEntryTypeQuery = new StringBuilder()
-                        .append("UPDATE ")
-                        .append(escape_Table("ENTRY"))
-                        .append(" SET ")
-                        .append(escape("TYPE"))
-                        .append(" = ?, ")
-                        .append(escape("VERSION"))
-                        .append(" = ")
-                        .append(escape("VERSION"))
-                        .append(" + 1 WHERE ")
-                        .append(escape("SHARED_ID"))
-                        .append(" = ?");
+                String updateEntryTypeQuery = "UPDATE " +
+                        escape_Table("ENTRY") +
+                        " SET " +
+                        escape("TYPE") +
+                        " = ?, " +
+                        escape("VERSION") +
+                        " = " +
+                        escape("VERSION") +
+                        " + 1 WHERE " +
+                        escape("SHARED_ID") +
+                        " = ?";
 
-                try (PreparedStatement preparedUpdateEntryTypeStatement = connection.prepareStatement(updateEntryTypeQuery.toString())) {
+                try (PreparedStatement preparedUpdateEntryTypeStatement = connection.prepareStatement(updateEntryTypeQuery)) {
                     preparedUpdateEntryTypeStatement.setString(1, localBibEntry.getType().getName());
                     preparedUpdateEntryTypeStatement.setInt(2, localBibEntry.getSharedBibEntryData().getSharedID());
                     preparedUpdateEntryTypeStatement.executeUpdate();
@@ -366,17 +360,16 @@ public abstract class DBMSProcessor {
         Set<Field> nullFields = new HashSet<>(sharedBibEntry.getFields());
         nullFields.removeAll(localBibEntry.getFields());
         for (Field nullField : nullFields) {
-            StringBuilder deleteFieldQuery = new StringBuilder()
-                    .append("DELETE FROM ")
-                    .append(escape_Table("FIELD"))
-                    .append(" WHERE ")
-                    .append(escape("NAME"))
-                    .append(" = ? AND ")
-                    .append(escape("ENTRY_SHARED_ID"))
-                    .append(" = ?");
+            String deleteFieldQuery = "DELETE FROM " +
+                    escape_Table("FIELD") +
+                    " WHERE " +
+                    escape("NAME") +
+                    " = ? AND " +
+                    escape("ENTRY_SHARED_ID") +
+                    " = ?";
 
             try (PreparedStatement preparedDeleteFieldStatement = connection
-                    .prepareStatement(deleteFieldQuery.toString())) {
+                    .prepareStatement(deleteFieldQuery)) {
                 preparedDeleteFieldStatement.setString(1, nullField.getName());
                 preparedDeleteFieldStatement.setInt(2, localBibEntry.getSharedBibEntryData().getSharedID());
                 preparedDeleteFieldStatement.executeUpdate();
@@ -397,54 +390,51 @@ public abstract class DBMSProcessor {
                 value = valueOptional.get();
             }
 
-            StringBuilder selectFieldQuery = new StringBuilder()
-                    .append("SELECT * FROM ")
-                    .append(escape_Table("FIELD"))
-                    .append(" WHERE ")
-                    .append(escape("NAME"))
-                    .append(" = ? AND ")
-                    .append(escape("ENTRY_SHARED_ID"))
-                    .append(" = ?");
+            String selectFieldQuery = "SELECT * FROM " +
+                    escape_Table("FIELD") +
+                    " WHERE " +
+                    escape("NAME") +
+                    " = ? AND " +
+                    escape("ENTRY_SHARED_ID") +
+                    " = ?";
 
             try (PreparedStatement preparedSelectFieldStatement = connection
-                    .prepareStatement(selectFieldQuery.toString())) {
+                    .prepareStatement(selectFieldQuery)) {
                 preparedSelectFieldStatement.setString(1, field.getName());
                 preparedSelectFieldStatement.setInt(2, localBibEntry.getSharedBibEntryData().getSharedID());
 
                 try (ResultSet selectFieldResultSet = preparedSelectFieldStatement.executeQuery()) {
                     if (selectFieldResultSet.next()) { // check if field already exists
-                        StringBuilder updateFieldQuery = new StringBuilder()
-                                .append("UPDATE ")
-                                .append(escape_Table("FIELD"))
-                                .append(" SET ")
-                                .append(escape("VALUE"))
-                                .append(" = ? WHERE ")
-                                .append(escape("NAME"))
-                                .append(" = ? AND ")
-                                .append(escape("ENTRY_SHARED_ID"))
-                                .append(" = ?");
+                        String updateFieldQuery = "UPDATE " +
+                                escape_Table("FIELD") +
+                                " SET " +
+                                escape("VALUE") +
+                                " = ? WHERE " +
+                                escape("NAME") +
+                                " = ? AND " +
+                                escape("ENTRY_SHARED_ID") +
+                                " = ?";
 
                         try (PreparedStatement preparedUpdateFieldStatement = connection
-                                .prepareStatement(updateFieldQuery.toString())) {
+                                .prepareStatement(updateFieldQuery)) {
                             preparedUpdateFieldStatement.setString(1, value);
                             preparedUpdateFieldStatement.setString(2, field.getName());
                             preparedUpdateFieldStatement.setInt(3, localBibEntry.getSharedBibEntryData().getSharedID());
                             preparedUpdateFieldStatement.executeUpdate();
                         }
                     } else {
-                        StringBuilder insertFieldQuery = new StringBuilder()
-                                .append("INSERT INTO ")
-                                .append(escape_Table("FIELD"))
-                                .append("(")
-                                .append(escape("ENTRY_SHARED_ID"))
-                                .append(", ")
-                                .append(escape("NAME"))
-                                .append(", ")
-                                .append(escape("VALUE"))
-                                .append(") VALUES(?, ?, ?)");
+                        String insertFieldQuery = "INSERT INTO " +
+                                escape_Table("FIELD") +
+                                "(" +
+                                escape("ENTRY_SHARED_ID") +
+                                ", " +
+                                escape("NAME") +
+                                ", " +
+                                escape("VALUE") +
+                                ") VALUES(?, ?, ?)";
 
                         try (PreparedStatement preparedFieldStatement = connection
-                                .prepareStatement(insertFieldQuery.toString())) {
+                                .prepareStatement(insertFieldQuery)) {
                             preparedFieldStatement.setInt(1, localBibEntry.getSharedBibEntryData().getSharedID());
                             preparedFieldStatement.setString(2, field.getName());
                             preparedFieldStatement.setString(3, value);
@@ -572,7 +562,7 @@ public abstract class DBMSProcessor {
 
                     // In all cases, we set the field value of the newly created BibEntry object
                     String value = selectEntryResultSet.getString("VALUE");
-                    if (value != null) {
+                    if (value != null && bibEntry != null) {
                         bibEntry.setField(FieldFactory.parseField(selectEntryResultSet.getString("NAME")), value, EntriesEventSource.SHARED);
                     }
                 }
@@ -595,13 +585,12 @@ public abstract class DBMSProcessor {
      */
     public Map<Integer, Integer> getSharedIDVersionMapping() {
         Map<Integer, Integer> sharedIDVersionMapping = new HashMap<>();
-        StringBuilder selectEntryQuery = new StringBuilder()
-                .append("SELECT * FROM ")
-                .append(escape_Table("ENTRY"))
-                .append(" ORDER BY ")
-                .append(escape("SHARED_ID"));
+        String selectEntryQuery = "SELECT * FROM " +
+                escape_Table("ENTRY") +
+                " ORDER BY " +
+                escape("SHARED_ID");
 
-        try (ResultSet selectEntryResultSet = connection.createStatement().executeQuery(selectEntryQuery.toString())) {
+        try (ResultSet selectEntryResultSet = connection.createStatement().executeQuery(selectEntryQuery)) {
             while (selectEntryResultSet.next()) {
                 sharedIDVersionMapping.put(selectEntryResultSet.getInt("SHARED_ID"), selectEntryResultSet.getInt("VERSION"));
             }

--- a/src/main/java/org/jabref/logic/shared/DBMSProcessor.java
+++ b/src/main/java/org/jabref/logic/shared/DBMSProcessor.java
@@ -62,7 +62,7 @@ public abstract class DBMSProcessor {
         if (type == DBMSType.POSTGRESQL || type == DBMSType.MYSQL) {
             String metadataVersion = metadata.get(MetaData.VERSION_DB_STRUCT);
             if (metadataVersion != null) {
-                int VERSION_DB_STRUCT = Integer.parseInt(metadata.getOrDefault(MetaData.VERSION_DB_STRUCT, ""));
+                int VERSION_DB_STRUCT = Integer.parseInt(metadata.getOrDefault(MetaData.VERSION_DB_STRUCT, "").replace(";", "");
                 if (VERSION_DB_STRUCT == getCURRENT_VERSION_DB_STRUCT()) {
                     databasePassesIntegrityCheck = true;
                 }

--- a/src/main/java/org/jabref/logic/shared/DBMSProcessor.java
+++ b/src/main/java/org/jabref/logic/shared/DBMSProcessor.java
@@ -62,7 +62,7 @@ public abstract class DBMSProcessor {
         if (type == DBMSType.POSTGRESQL || type == DBMSType.MYSQL) {
             String metadataVersion = metadata.get(MetaData.VERSION_DB_STRUCT);
             if (metadataVersion != null) {
-                int VERSION_DB_STRUCT = Integer.parseInt(metadata.getOrDefault(MetaData.VERSION_DB_STRUCT, "").replace(";", "");
+                int VERSION_DB_STRUCT = Integer.parseInt(metadata.getOrDefault(MetaData.VERSION_DB_STRUCT, "").replace(";" ,""));
                 if (VERSION_DB_STRUCT == getCURRENT_VERSION_DB_STRUCT()) {
                     databasePassesIntegrityCheck = true;
                 }

--- a/src/main/java/org/jabref/logic/shared/DBMSProcessor.java
+++ b/src/main/java/org/jabref/logic/shared/DBMSProcessor.java
@@ -62,7 +62,7 @@ public abstract class DBMSProcessor {
         if (type == DBMSType.POSTGRESQL || type == DBMSType.MYSQL) {
             String metadataVersion = metadata.get(MetaData.VERSION_DB_STRUCT);
             if (metadataVersion != null) {
-                int VERSION_DB_STRUCT = Integer.parseInt(metadata.getOrDefault(MetaData.VERSION_DB_STRUCT,"").replace(";",""));
+                int VERSION_DB_STRUCT = Integer.parseInt(metadata.getOrDefault(MetaData.VERSION_DB_STRUCT, "").replace(";",""));
                 if (VERSION_DB_STRUCT == getCURRENT_VERSION_DB_STRUCT()) {
                     databasePassesIntegrityCheck = true;
                 }

--- a/src/main/java/org/jabref/logic/shared/DBMSProcessor.java
+++ b/src/main/java/org/jabref/logic/shared/DBMSProcessor.java
@@ -60,9 +60,12 @@ public abstract class DBMSProcessor {
         DBMSType type = this.connectionProperties.getType();
         Map<String, String> metadata = getSharedMetaData();
         if (type == DBMSType.POSTGRESQL || type == DBMSType.MYSQL) {
-            int VERSION_DB_STRUCT = Integer.parseInt(metadata.get(MetaData.VERSION_DB_STRUCT));
-            if (VERSION_DB_STRUCT == getCURRENT_VERSION_DB_STRUCT()) {
-                databasePassesIntegrityCheck = true;
+            String metadataVersion = metadata.get(MetaData.VERSION_DB_STRUCT);
+            if (metadataVersion != null) {
+                int VERSION_DB_STRUCT = Integer.parseInt(metadata.getOrDefault(MetaData.VERSION_DB_STRUCT, ""), );
+                if (VERSION_DB_STRUCT == getCURRENT_VERSION_DB_STRUCT()) {
+                    databasePassesIntegrityCheck = true;
+                }
             }
         } else {
             databasePassesIntegrityCheck = checkTableAvailability("ENTRY", "FIELD", "METADATA");

--- a/src/main/java/org/jabref/logic/shared/DBMSProcessor.java
+++ b/src/main/java/org/jabref/logic/shared/DBMSProcessor.java
@@ -62,7 +62,7 @@ public abstract class DBMSProcessor {
         if (type == DBMSType.POSTGRESQL || type == DBMSType.MYSQL) {
             String metadataVersion = metadata.get(MetaData.VERSION_DB_STRUCT);
             if (metadataVersion != null) {
-                int VERSION_DB_STRUCT = Integer.parseInt(metadata.getOrDefault(MetaData.VERSION_DB_STRUCT, "").replace(";",""));
+                int VERSION_DB_STRUCT = Integer.parseInt(metadata.getOrDefault(MetaData.VERSION_DB_STRUCT, "").replace(";", ""));
                 if (VERSION_DB_STRUCT == getCURRENT_VERSION_DB_STRUCT()) {
                     databasePassesIntegrityCheck = true;
                 }

--- a/src/main/java/org/jabref/logic/shared/DBMSProcessor.java
+++ b/src/main/java/org/jabref/logic/shared/DBMSProcessor.java
@@ -62,7 +62,7 @@ public abstract class DBMSProcessor {
         if (type == DBMSType.POSTGRESQL || type == DBMSType.MYSQL) {
             String metadataVersion = metadata.get(MetaData.VERSION_DB_STRUCT);
             if (metadataVersion != null) {
-                int VERSION_DB_STRUCT = Integer.parseInt(metadata.getOrDefault(MetaData.VERSION_DB_STRUCT, ""), );
+                int VERSION_DB_STRUCT = Integer.parseInt(metadata.getOrDefault(MetaData.VERSION_DB_STRUCT, ""));
                 if (VERSION_DB_STRUCT == getCURRENT_VERSION_DB_STRUCT()) {
                     databasePassesIntegrityCheck = true;
                 }
@@ -185,9 +185,7 @@ public abstract class DBMSProcessor {
                 .append(escape("TYPE"))
                 .append(") VALUES(?)");
         // Number of commas is bibEntries.size() - 1
-        for (int i = 0; i < (bibEntries.size() - 1); i++) {
-            insertIntoEntryQuery.append(", (?)");
-        }
+        insertIntoEntryQuery.append(", (?)".repeat(Math.max(0, (bibEntries.size() - 1))));
 
         try (PreparedStatement preparedEntryStatement = connection.prepareStatement(insertIntoEntryQuery.toString(),
                 new String[]{"SHARED_ID"})) {
@@ -278,9 +276,7 @@ public abstract class DBMSProcessor {
             }
 
             // Number of commas is fields.size() - 1
-            for (int i = 0; i < (numFields - 1); i++) {
-                insertFieldQuery.append(", (?, ?, ?)");
-            }
+            insertFieldQuery.append(", (?, ?, ?)".repeat(Math.max(0, (numFields - 1))));
             try (PreparedStatement preparedFieldStatement = connection.prepareStatement(insertFieldQuery.toString())) {
                 int fieldsCompleted = 0;
                 for (int entryIndex = 0; entryIndex < fields.size(); entryIndex++) {
@@ -571,7 +567,7 @@ public abstract class DBMSProcessor {
                 }
             }
         } catch (SQLException e) {
-            LOGGER.error("Executed >{}<", query.toString());
+            LOGGER.error("Executed >{}<", query);
             LOGGER.error("SQL Error", e);
             return Collections.emptyList();
         }

--- a/src/main/java/org/jabref/logic/shared/DBMSProcessor.java
+++ b/src/main/java/org/jabref/logic/shared/DBMSProcessor.java
@@ -62,7 +62,7 @@ public abstract class DBMSProcessor {
         if (type == DBMSType.POSTGRESQL || type == DBMSType.MYSQL) {
             String metadataVersion = metadata.get(MetaData.VERSION_DB_STRUCT);
             if (metadataVersion != null) {
-                int VERSION_DB_STRUCT = Integer.parseInt(metadata.getOrDefault(MetaData.VERSION_DB_STRUCT, "").replace(";" ,""));
+                int VERSION_DB_STRUCT = Integer.parseInt(metadata.getOrDefault(MetaData.VERSION_DB_STRUCT,"").replace(";",""));
                 if (VERSION_DB_STRUCT == getCURRENT_VERSION_DB_STRUCT()) {
                     databasePassesIntegrityCheck = true;
                 }

--- a/src/main/java/org/jabref/logic/shared/PostgreSQLProcessor.java
+++ b/src/main/java/org/jabref/logic/shared/PostgreSQLProcessor.java
@@ -21,8 +21,8 @@ public class PostgreSQLProcessor extends DBMSProcessor {
 
     private PostgresSQLNotificationListener listener;
 
-    private Integer VERSION_DB_STRUCT_DEFAULT = -1;
-    private Integer CURRENT_VERSION_DB_STRUCT = 1;
+    private int VERSION_DB_STRUCT_DEFAULT = -1;
+    private final int CURRENT_VERSION_DB_STRUCT = 1;
 
     public PostgreSQLProcessor(DatabaseConnection connection) {
         super(connection);
@@ -31,7 +31,7 @@ public class PostgreSQLProcessor extends DBMSProcessor {
     /**
      * Creates and sets up the needed tables and columns according to the database type.
      *
-     * @throws SQLException
+     * @throws SQLException in case of error
      */
     @Override
     public void setUp() throws SQLException {
@@ -65,7 +65,8 @@ public class PostgreSQLProcessor extends DBMSProcessor {
 
         if (metadata.get(MetaData.VERSION_DB_STRUCT) != null) {
             try {
-                VERSION_DB_STRUCT_DEFAULT = Integer.valueOf(metadata.get(MetaData.VERSION_DB_STRUCT));
+                // replace semicolon so we can parse it
+                VERSION_DB_STRUCT_DEFAULT = Integer.parseInt(metadata.get(MetaData.VERSION_DB_STRUCT).replace(";", ""));
             } catch (Exception e) {
                 LOGGER.warn("[VERSION_DB_STRUCT_DEFAULT] not Integer!");
             }
@@ -84,7 +85,7 @@ public class PostgreSQLProcessor extends DBMSProcessor {
                 metadata = getSharedMetaData();
             }
 
-            metadata.put(MetaData.VERSION_DB_STRUCT, CURRENT_VERSION_DB_STRUCT.toString());
+            metadata.put(MetaData.VERSION_DB_STRUCT, String.valueOf(CURRENT_VERSION_DB_STRUCT));
             setSharedMetaData(metadata);
         }
     }
@@ -98,9 +99,7 @@ public class PostgreSQLProcessor extends DBMSProcessor {
                 .append(escape("TYPE"))
                 .append(") VALUES(?)");
         // Number of commas is bibEntries.size() - 1
-        for (int i = 0; i < bibEntries.size() - 1; i++) {
-            insertIntoEntryQuery.append(", (?)");
-        }
+        insertIntoEntryQuery.append(", (?)".repeat(Math.max(0, bibEntries.size() - 1)));
         try (PreparedStatement preparedEntryStatement = connection.prepareStatement(insertIntoEntryQuery.toString(),
                 Statement.RETURN_GENERATED_KEYS)) {
             for (int i = 0; i < bibEntries.size(); i++) {

